hls4ml/backends/fpga/passes/clone.py

```diff
@@ -64,13 +65,20 @@ def transform(self, model, node):
                     print('WARNING: Cloning output {} of {} ({}) more than 3 times not currently supported'.format(output, node.__class__.__name__, node.name))
                     return False
                 out_var = node.get_output_variable(output)
+                attrs = {
+                    'size' : np.prod(out_var.shape)
+                }
+                clone_layer = model.make_node(Clone, 'clone_' + node.name, attrs, [output], [output + '_cpy' + str(i + 1) for i in range(len(output_map[output]))])
                 for i, layer in enumerate(output_map[output], 1):
-                    attrs = {
-                        'size' : np.prod(out_var.shape)
-                    }
                     idx = layer.inputs.index(output)
                     layer.inputs[idx] = output + '_cpy' + str(i)
-                clone_layer = model.make_node(Clone, 'clone_' + node.name, attrs, [output], [output + '_cpy' + str(i + 1) for i in range(len(output_map[output]))])
+                    if isinstance(layer, Reshape):
```

Comment:
This is band-aid for the fundamental problem of InplaceVariables, and basically every optimizer would need to include it. The issue is not just here. This is what I attempted to solve in #525, but it subsequently got more work done in #591. I think that's the real solution.

Reply:
Yes, I see your point.

Reply:
OK removed from here now